<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="daiuc" />

<meta name="date" content="2021-08-17" />

<title>ldsc_run</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lab3updates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lab3updates/lab3updates">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ldsc_run</h1>
<h4 class="author">daiuc</h4>
<h4 class="date">2021-08-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lab3updates/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210810code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210810)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210810code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210810)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdaiuclab3updatestree3b937d4f23f438ce2751c05ff179d925fa3f6398targetblank3b937d4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/daiuc/lab3updates/tree/3b937d4f23f438ce2751c05ff179d925fa3f6398" target="_blank">3b937d4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomdaiuclab3updatestree3b937d4f23f438ce2751c05ff179d925fa3f6398targetblank3b937d4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/daiuc/lab3updates/tree/3b937d4f23f438ce2751c05ff179d925fa3f6398" target="_blank">3b937d4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ldsc_run.Rmd</code>) and HTML (<code>docs/ldsc_run.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/3b937d4f23f438ce2751c05ff179d925fa3f6398/docs/ldsc_run.html" target="_blank">3b937d4</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/cb7a46e8fb3f3854328f92898abc55ec9a8b73a2/analysis/ldsc_run.Rmd" target="_blank">cb7a46e</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
<td>
updated for RCC; used CDS length, GTEx signif SNPS slopes are now z-scores
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/cb7a46e8fb3f3854328f92898abc55ec9a8b73a2/docs/ldsc_run.html" target="_blank">cb7a46e</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
<td>
updated for RCC; used CDS length, GTEx signif SNPS slopes are now z-scores
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/abf344d658273f4dc214a220408aad8335c945c0/docs/ldsc_run.html" target="_blank">abf344d</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/f497973d6c17e969900e525e555267e000857893/analysis/ldsc_run.Rmd" target="_blank">f497973</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/84c61eb4eccaca986ae094488701b92ddf00531c/docs/ldsc_run.html" target="_blank">84c61eb</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/8d7a134428ac24e07d5b0828d356f0eda09f76db/analysis/ldsc_run.Rmd" target="_blank">8d7a134</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/19bfb03b0fa0e1577f7495f497ea0ad9a813ce3c/docs/ldsc_run.html" target="_blank">19bfb03</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/dcb2524df49dd1431377b84ba70dab2b75e42092/analysis/ldsc_run.Rmd" target="_blank">dcb2524</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/dcb2524df49dd1431377b84ba70dab2b75e42092/docs/ldsc_run.html" target="_blank">dcb2524</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_publish(all = T)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/b9363766b087f6d331093379f35edd05c7adda16/analysis/ldsc_run.Rmd" target="_blank">b936376</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_git_commit(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/b9363766b087f6d331093379f35edd05c7adda16/docs/ldsc_run.html" target="_blank">b936376</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_git_commit(all = T)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/6d54ca01aa92f44ea95a829766552ad19ec82e3f/analysis/ldsc_run.Rmd" target="_blank">6d54ca0</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_git_commit(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/6d54ca01aa92f44ea95a829766552ad19ec82e3f/docs/ldsc_run.html" target="_blank">6d54ca0</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_git_commit(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/5c7697fd4f138e4c8778d51ab4b4ff2475b1bfda/docs/ldsc_run.html" target="_blank">5c7697f</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/55849935cbf749856c737135b69c85c1256ab0de/analysis/ldsc_run.Rmd" target="_blank">5584993</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/a1855cad6cfc36faef3399ba0eb68f2aef8a15f3/docs/ldsc_run.html" target="_blank">a1855ca</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/d749e6787cf92dcc1b7a420aed22f8f2035ead1f/analysis/ldsc_run.Rmd" target="_blank">d749e67</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/2662304af90a7356385b932986ab2bfd1c8e33b1/docs/ldsc_run.html" target="_blank">2662304</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/9efd8260a50370ab1e34e23b7dfabfdcdf79aca0/analysis/ldsc_run.Rmd" target="_blank">9efd826</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/1392841f108c62c85f6f4e63e9afe31568a4bb8e/docs/ldsc_run.html" target="_blank">1392841</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/090c2150c45afd6ec9ba499c9fa3ae93f1c1550c/analysis/ldsc_run.Rmd" target="_blank">090c215</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/88193d649a43857c87c261c304bbe24bcb59aaac/docs/ldsc_run.html" target="_blank">88193d6</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/e11a24407f9b7aa33b399050420708ea4252e280/analysis/ldsc_run.Rmd" target="_blank">e11a244</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/f534edc5c7bdf8ad1712495f1f7b848df8457172/analysis/ldsc_run.Rmd" target="_blank">f534edc</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/f534edc5c7bdf8ad1712495f1f7b848df8457172/docs/ldsc_run.html" target="_blank">f534edc</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/61bffed7486b76cf6221d4d02fd4a11dbade2a7c/docs/ldsc_run.html" target="_blank">61bffed</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/572fceaf13008e326df55c98c6a807714e2bfffd/analysis/ldsc_run.Rmd" target="_blank">572fcea</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”, force = T, update = T,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/1643aee6ed342b1e3908d4f5e172e34afe78002a/docs/ldsc_run.html" target="_blank">1643aee</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/63b24a3149b5ee13766024ebb13bf2819eb717a4/analysis/ldsc_run.Rmd" target="_blank">63b24a3</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”, force = T, update = T,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/a3470abe9b08428ab21bf44380d49e7b397566fb/docs/ldsc_run.html" target="_blank">a3470ab</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/780076578ded5d41bb6f87a2e94cac146c0c7d90/docs/ldsc_run.html" target="_blank">7800765</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/78b116b0eaabfc560db9dbf19ca88b8c1351b3d6/analysis/ldsc_run.Rmd" target="_blank">78b116b</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”, force = T, update = T,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/9cdac13c1178a2d4b945f54f1d655c037d841c22/docs/ldsc_run.html" target="_blank">9cdac13</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/91add15fe7824f6b2484f48a9cba3e435aa1827f/docs/ldsc_run.html" target="_blank">91add15</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/b4016e65401e1ee408a79ff263a2de87aabd6452/analysis/ldsc_run.Rmd" target="_blank">b4016e6</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(“analysis/ldsc_run.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/36fb6e0e3e9e3ff1e892eebcdc3d31f9cf3a79d6/docs/ldsc_run.html" target="_blank">36fb6e0</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/6c43157bc5dc4ce43b5982fb738b09ca780bfe70/analysis/ldsc_run.Rmd" target="_blank">6c43157</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(c(“analysis/modified_code.Rmd”, “analysis/ldsc_run.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/f11e618e0e039ea7fba9e5c30adebe02a6b44eef/docs/ldsc_run.html" target="_blank">f11e618</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/386560e67d04e7fcc3ab0c62a5117a697018d838/docs/ldsc_run.html" target="_blank">386560e</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/c5ff00f80ae4c761ec0b56cecc238b1780c59dac/analysis/ldsc_run.Rmd" target="_blank">c5ff00f</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(c(“analysis/modified_code.Rmd”, “analysis/ldsc_run.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/cf145bb76ed21e1449d5707e252ef8b316a6bbcd/docs/ldsc_run.html" target="_blank">cf145bb</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/39b918d089a2e6c064c56b3f56a597bebc65bfa4/analysis/ldsc_run.Rmd" target="_blank">39b918d</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
wflow_publish(c(“analysis/modified_code.Rmd”, “analysis/ldsc_run.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/470d29cc8784ed949bfdb9236b92044c59286b59/analysis/ldsc_run.Rmd" target="_blank">470d29c</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/470d29cc8784ed949bfdb9236b92044c59286b59/docs/ldsc_run.html" target="_blank">470d29c</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/74cc100c981a2a7fd95fae631d77f64ffedf1555/analysis/ldsc_run.Rmd" target="_blank">74cc100</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
a
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/74cc100c981a2a7fd95fae631d77f64ffedf1555/docs/ldsc_run.html" target="_blank">74cc100</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
a
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/f505867f410d208654704cdcd7021ca15ec2b092/docs/ldsc_run.html" target="_blank">f505867</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/f1dc1316ce982f1585ae3fa34219626a61c76f07/docs/ldsc_run.html" target="_blank">f1dc131</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/e5f7d680129220bcf85c325926484ea51e0209d9/analysis/ldsc_run.Rmd" target="_blank">e5f7d68</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
wflow_publish(c(“analysis/ldsc_run.Rmd”))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/1027954b5764a8436430f4e77cdb90fc5bc5fb20/analysis/ldsc_run.Rmd" target="_blank">1027954</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
edit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/daiuc/lab3updates/1027954b5764a8436430f4e77cdb90fc5bc5fb20/docs/ldsc_run.html" target="_blank">1027954</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-17
</td>
<td>
edit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective-1-estimate-each-genes-total-cis-heritability" class="section level2">
<h2>Objective 1: estimate each gene’s total cis heritability</h2>
<p>A given gene’s total heritability can be estimated by first estimate average per SNP heritability for each annotation category <span class="math inline">\(\bar{\tau_c}\)</span>. Then A given gene’s total heritability can be computed by multiplying number of all qualifying SNPs in cis Modified <code>ldsc</code> codes to output <span class="math inline">\(\tau_c\)</span> of each category as well as overlapping SNPs.</p>
<p><span class="math display">\[h^2_g(total) = \sum_C{M_c \bar{\tau_c}}\]</span></p>
<p>To compute <span class="math inline">\(M_c\)</span> and <span class="math inline">\(\bar{\tau_c}\)</span>, I made these modifications in the original scripts of <a href="https://github.com/bulik/ldsc">LD SCore v1.0.1</a>:</p>
<ul>
<li><p>Output number of SNPs for each category, essentially an inner join between cis-eQTL summary stats SNPs and LD score SNPs from each of the 53 category (including base, ie. just SNPs).</p></li>
<li><p>This involves modifying the functions of:</p>
<ul>
<li><code>annot_parser</code>, <code>annot</code> functions in the <code>parse.py</code> script</li>
<li><code>_read_annot</code> function in the <code>sumstats.py</code> script</li>
<li><code>_overlap_output</code> method in the <code>regression.py</code> script</li>
</ul></li>
<li><p>With the modification, <code>ldsc</code> will now output correct intersected <code>M_annot</code> or number of cis-eQTL SNPs for each category</p></li>
<li><p>Code for computing coefficient is not modified, since it already only take in intersection between ld score SNPs and cis-eQTL SNPs</p></li>
</ul>
<p>Result: - <span class="math inline">\(h^2\)</span> for 18373 genes are estimated</p>
<p><a href="modified_code.html">Modified code</a></p>
<hr />
</div>
<div id="objective-2-explore-the-relationships-between-heritability-and-gene-features" class="section level2">
<h2>Objective 2: explore the relationships between heritability and gene features</h2>
<p>Effectively, we now compare cis gene heritability estimated from above with other metrics that measure regulatory or functional characteristics of genomic regions / genes.</p>
<p>The following dataset from Wang et al. AJHG 2020 (<a href="https://doi.org/10.1016/j.ajhg.2020.01.012">link</a>) are used. <code>15,913</code> genes are incorporated from this dataset (after removing NA values). The <code>h^2</code> dataset and teh <code>EDS</code> dataset are joined together on gene IDs. After having removed negative <code>h^2</code> genes, <code>8,936</code> genes remain. Zscores are computed for each gene.</p>
<p><br></p>
<div id="example-of-combined-dataset" class="section level3">
<h3>Example of combined dataset</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
gene_id
</th>
<th style="text-align:left;">
gene_nm
</th>
<th style="text-align:right;">
h2
</th>
<th style="text-align:right;">
m_tot
</th>
<th style="text-align:right;">
EDS
</th>
<th style="text-align:right;">
pLI
</th>
<th style="text-align:right;">
RVIS
</th>
<th style="text-align:right;">
max.abs.slope_z
</th>
<th style="text-align:right;">
N_QTLs
</th>
<th style="text-align:right;">
length
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000000419
</td>
<td style="text-align:left;">
DPM1
</td>
<td style="text-align:right;">
0.0105250
</td>
<td style="text-align:right;">
576
</td>
<td style="text-align:right;">
0.4441988
</td>
<td style="text-align:right;">
0.5367475
</td>
<td style="text-align:right;">
64.45
</td>
<td style="text-align:right;">
4.595151
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
3187
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000000457
</td>
<td style="text-align:left;">
SCYL3
</td>
<td style="text-align:right;">
0.0179916
</td>
<td style="text-align:right;">
381
</td>
<td style="text-align:right;">
0.5631780
</td>
<td style="text-align:right;">
0.5470538
</td>
<td style="text-align:right;">
51.26
</td>
<td style="text-align:right;">
9.650809
</td>
<td style="text-align:right;">
1358
</td>
<td style="text-align:right;">
8282
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000000460
</td>
<td style="text-align:left;">
C1orf112
</td>
<td style="text-align:right;">
0.0092650
</td>
<td style="text-align:right;">
354
</td>
<td style="text-align:right;">
0.2556235
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
10.81
</td>
<td style="text-align:right;">
12.610171
</td>
<td style="text-align:right;">
1564
</td>
<td style="text-align:right;">
8235
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000000938
</td>
<td style="text-align:left;">
FGR
</td>
<td style="text-align:right;">
0.0079872
</td>
<td style="text-align:right;">
207
</td>
<td style="text-align:right;">
0.4975906
</td>
<td style="text-align:right;">
0.1771744
</td>
<td style="text-align:right;">
11.68
</td>
<td style="text-align:right;">
7.325070
</td>
<td style="text-align:right;">
194
</td>
<td style="text-align:right;">
7286
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000000971
</td>
<td style="text-align:left;">
CFH
</td>
<td style="text-align:right;">
0.0044361
</td>
<td style="text-align:right;">
249
</td>
<td style="text-align:right;">
0.4242650
</td>
<td style="text-align:right;">
0.9996310
</td>
<td style="text-align:right;">
39.03
</td>
<td style="text-align:right;">
9.885834
</td>
<td style="text-align:right;">
477
</td>
<td style="text-align:right;">
6195
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000001036
</td>
<td style="text-align:left;">
FUCA2
</td>
<td style="text-align:right;">
0.0100081
</td>
<td style="text-align:right;">
436
</td>
<td style="text-align:right;">
0.5437246
</td>
<td style="text-align:right;">
0.0003656
</td>
<td style="text-align:right;">
57.48
</td>
<td style="text-align:right;">
5.587449
</td>
<td style="text-align:right;">
186
</td>
<td style="text-align:right;">
1728
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="qc-plots" class="section level3">
<h3>QC plots</h3>
<p>After rounds of filtering about 9000 protein coding genes are left for downstream analysis</p>
<p><br></p>
<div id="summary-stats-of-qualifying-snps-per-gene" class="section level4">
<h4>Summary stats of qualifying SNPs per gene</h4>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   12.0   352.0   445.0   448.6   545.0  1139.0 </code></pre>
<p><br></p>
</div>
<div id="distribution-of-qualifying-snps-for-h2-estimate" class="section level4">
<h4>Distribution of qualifying SNPs for <span class="math inline">\(h^2\)</span> estimate</h4>
<p><img src="figure/ldsc_run.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/cb7a46e8fb3f3854328f92898abc55ec9a8b73a2/docs/figure/ldsc_run.Rmd/unnamed-chunk-9-1.png" target="_blank">cb7a46e</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/5c7697fd4f138e4c8778d51ab4b4ff2475b1bfda/docs/figure/ldsc_run.Rmd/unnamed-chunk-9-1.png" target="_blank">5c7697f</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/f534edc5c7bdf8ad1712495f1f7b848df8457172/docs/figure/ldsc_run.Rmd/unnamed-chunk-9-1.png" target="_blank">f534edc</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/91add15fe7824f6b2484f48a9cba3e435aa1827f/docs/figure/ldsc_run.Rmd/unnamed-chunk-9-1.png" target="_blank">91add15</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h2-distribution" class="section level4">
<h4><span class="math inline">\(h^2\)</span> distribution</h4>
<p><img src="figure/ldsc_run.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/3b937d4f23f438ce2751c05ff179d925fa3f6398/docs/figure/ldsc_run.Rmd/unnamed-chunk-10-1.png" target="_blank">3b937d4</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="test-correlation-between-qualifying-snps-and-heritability." class="section level4">
<h4>Test correlation between qualifying SNPs and heritability.</h4>
<p>More qualifying SNPs do seem to contribute to higher heribability. However, longer genes do not seem to include more qualifying SNPs.</p>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  m_tot and h2
t = 30.976, df = 8936, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.2925448 0.3299896
sample estimates:
      cor 
0.3113881 </code></pre>
<p><img src="figure/ldsc_run.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/3b937d4f23f438ce2751c05ff179d925fa3f6398/docs/figure/ldsc_run.Rmd/unnamed-chunk-12-1.png" target="_blank">3b937d4</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/19bfb03b0fa0e1577f7495f497ea0ad9a813ce3c/docs/figure/ldsc_run.Rmd/unnamed-chunk-12-1.png" target="_blank">19bfb03</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/5c7697fd4f138e4c8778d51ab4b4ff2475b1bfda/docs/figure/ldsc_run.Rmd/unnamed-chunk-12-1.png" target="_blank">5c7697f</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/f534edc5c7bdf8ad1712495f1f7b848df8457172/docs/figure/ldsc_run.Rmd/unnamed-chunk-12-1.png" target="_blank">f534edc</a>
</td>
<td>
daiuc
</td>
<td>
2021-08-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><br></p>
</div>
</div>
<div id="is-cis-genetic-heritability-related-to-regulatory-landscape-functional-intolerance-and-constraints-or-gene-length" class="section level3">
<h3>Is cis genetic heritability related to regulatory landscape, functional intolerance and constraints, or gene length?</h3>
<p>To answer this question, I investigated the following metrics:</p>
<p><strong>EDS</strong></p>
<p>EDS, or enhancer-domain score, reflects the total size and redundancy of a gene’s non-coding regulatory architecture. <em>A higher EDS suggests a gene has higher non-coding regulatory interactions</em>. It was noted to be negatively correlated with cis eQTL (<a href="https://doi.org/10.1016/j.ajhg.2020.01.012">Wang et al. 2020</a>). EDS is computed by first constructing enhancer domains for each gene from enhancer predictions from various data sources. For each enhancer domain per gene, domain size and redundancy of TF motifs are the main determinants of the final EDS value.</p>
<p>In my analysis, EDS data is downloaded from Wang et al. 2020 paper.</p>
<p><br></p>
<p><strong>RVIS</strong></p>
<p>RVIS (Residual Variation Intolerance Score) is designed to rank genes in terms of whether they have more or less common functional genetic variation <em>relative to the genome wide expectation</em> given the amount of apparently neutral variation the gene has. A gene with a positive score has more common functional variation, and a gene with a negative score has less and is referred to as “intolerant”. By convention we rank all genes in order from most intolerant to least. As an example, a gene such as ATP1A3 has a RVIS score of -1.53 and a percentile of 3.37%, meaning it is amongst the 3.37% most intolerant of human genes. <em>A negative score indicates a gene has less than expected amount of functional variants, thus may be under purifying selection, while a positive score indicates higher than expected amount of functional variants, thus may be under balanced or positive selection.</em></p>
<p>RVIS score in this analysis is downloaded from the Wang et al. 2020 paper.</p>
<p><strong>pLI</strong></p>
<p>pLI (probability of being loss-of-function intolerant) reflects the tolerance of a gene to the loss of function mutations. <em>A higher value of pLI indicates a gene’s function is more likely to be abrogated under a loss-of-function mutation</em>. It is frequently used to prioritize candidate genes when analyzing whole exome or whole genome data. Note that pLI specifically measures the impact of loss of function variants, such as missense, frame shift, early stop codon, etc.</p>
<p><strong>gene length</strong></p>
<p>I filtered protein coding genes and computed gene length from Gencode hg38 V31 GTF file.</p>
<p><strong>max.abs.slope</strong></p>
<p>I downloaded cis-eQTL significant SNP-gene pair dataset from GTEx V8 analysis. Data from 49 single tissue specific datasets are merged. For this metric as well as the <code>N_QTL</code> metric, all significant SNPs are subject to <code>MAF</code> filtering consistent with heritability estimate (<code>0.05 &lt; MAF &lt; 0.95</code>). Maximum effect size is summarised for each gene using the regression slope (<code>slope</code>) column in the <code>*.eqtl_signifpairs.txt.gz</code> data files. The <code>slope</code> values are first taken as absolute value before max summarising by gene.</p>
<p><strong>GTEx V8 N_QTL</strong></p>
<p>N_QTL refers to the number of significant cis-eQTLs per gene.</p>
<div id="distributions-of-other-gene-metrics" class="section level4">
<h4>Distributions of other gene metrics</h4>
<p><img src="figure/ldsc_run.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/3b937d4f23f438ce2751c05ff179d925fa3f6398/docs/figure/ldsc_run.Rmd/unnamed-chunk-15-1.png" target="_blank">3b937d4</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><br></p>
</div>
<div id="can-we-observe-correlation-between-h2-and-the-other-metrics" class="section level4">
<h4>Can we observe correlation between <span class="math inline">\(h^2\)</span> and the other metrics?</h4>
<p><br></p>
<p><img src="figure/ldsc_run.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/3b937d4f23f438ce2751c05ff179d925fa3f6398/docs/figure/ldsc_run.Rmd/unnamed-chunk-17-1.png" target="_blank">3b937d4</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><br></p>
</div>
<div id="demonstrating-correlation-between-h2-and-other-metrics" class="section level4">
<h4>Demonstrating correlation between <span class="math inline">\(h^2\)</span> and other metrics</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
p
</th>
<th style="text-align:right;">
cor.spearman
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
EDS
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
-0.1656608
</td>
</tr>
<tr>
<td style="text-align:left;">
pLI
</td>
<td style="text-align:right;">
0.0002150
</td>
<td style="text-align:right;">
-0.0401135
</td>
</tr>
<tr>
<td style="text-align:left;">
RVIS
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0802491
</td>
</tr>
<tr>
<td style="text-align:left;">
length
</td>
<td style="text-align:right;">
0.0873676
</td>
<td style="text-align:right;">
-0.0180826
</td>
</tr>
<tr>
<td style="text-align:left;">
max.abs.slope_z
</td>
<td style="text-align:right;">
0.5699956
</td>
<td style="text-align:right;">
-0.0060094
</td>
</tr>
<tr>
<td style="text-align:left;">
N_QTLs
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
-0.1338915
</td>
</tr>
</tbody>
</table>
<p><br></p>
<p><img src="figure/ldsc_run.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/daiuc/lab3updates/blob/3b937d4f23f438ce2751c05ff179d925fa3f6398/docs/figure/ldsc_run.Rmd/unnamed-chunk-21-1.png" target="_blank">3b937d4</a>
</td>
<td>
Chao Dai
</td>
<td>
2021-08-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><br></p>
</div>
<div id="interprete-result" class="section level4">
<h4>Interprete result</h4>
<p>I performed correlation test between <span class="math inline">\(h^2\)</span> and all the other metrics. <code>EDS</code>, <code>gene length</code>, and <code>N_eQTLs</code> showed the strongest correlation. All are inversely correlated with <span class="math inline">\(h^2\)</span>.</p>
<ul>
<li><p><code>EDS</code> was previously noted to negatively correlate with cis-eQTLs. One explanation is that mammalian genes have evolved a robustness to cope with regulatory genetic variation (Wang et al.). Thus genes with high EDS would be better explained by genomewide variants, rather than cis variants. This perhaps explains why higher EDS genes might have lower cis <span class="math inline">\(h^2\)</span>.</p></li>
<li><p>gene length:</p>
<ul>
<li>possibly because longer genes do not necessarily contributes to more qualifying SNPs, as a result longer genes are penalized for heritability?</li>
<li>should we restrict to length of CDS only?</li>
</ul></li>
<li><p>N_QTLs: strange that it’s negatively correlated.</p>
<ul>
<li>Perhaps due to many more SNPs/eQTLs in the GTEx dataset relative to SNPs used for <span class="math inline">\(h^2\)</span> (eQTLgen x EUR hapmap3)</li>
<li>eQTLgen only has blood samples, but GTEx more diverse</li>
</ul></li>
<li><p><code>RVIS</code> is positively correlated. Since a higher positive <code>RVIS</code> score means that there are likely more functional variants than expected (possible positive selection). Also from previous plot, it suggests that higher cis heritability may originate from the collective contribution of larger amount of common variants. From this standpoint, it makes sense that the two metrics should move in the same direction.</p></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.2 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] furrr_0.2.3       future_1.21.0     cowplot_1.1.1     data.table_1.14.0
 [5] forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7       purrr_0.3.4      
 [9] readr_2.0.1       tidyr_1.1.3       tibble_3.1.4      ggplot2_3.3.5    
[13] tidyverse_1.3.1   workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] nlme_3.1-152      fs_1.5.0          lubridate_1.7.10  webshot_0.5.2    
 [5] httr_1.4.2        rprojroot_2.0.2   tools_4.1.0       backports_1.2.1  
 [9] bslib_0.2.5.1     utf8_1.2.2        R6_2.5.1          DBI_1.1.1        
[13] mgcv_1.8-36       colorspace_2.0-2  withr_2.4.2       tidyselect_1.1.1 
[17] gridExtra_2.3     compiler_4.1.0    git2r_0.28.0      cli_3.0.1        
[21] rvest_1.0.1       xml2_1.3.2        labeling_0.4.2    sass_0.4.0       
[25] scales_1.1.1      systemfonts_1.0.2 digest_0.6.27     rmarkdown_2.10   
[29] svglite_2.0.0     pkgconfig_2.0.3   htmltools_0.5.1.1 parallelly_1.27.0
[33] dbplyr_2.1.1      highr_0.9         rlang_0.4.11      readxl_1.3.1     
[37] rstudioapi_0.13   jquerylib_0.1.4   generics_0.1.0    farver_2.1.0     
[41] jsonlite_1.7.2    magrittr_2.0.1    kableExtra_1.3.4  Matrix_1.3-4     
[45] Rcpp_1.0.7        munsell_0.5.0     fansi_0.5.0       lifecycle_1.0.0  
[49] stringi_1.7.3     whisker_0.4       yaml_2.2.1        grid_4.1.0       
[53] parallel_4.1.0    listenv_0.8.0     promises_1.2.0.1  crayon_1.4.1     
[57] lattice_0.20-44   haven_2.4.3       splines_4.1.0     hms_1.1.0        
[61] knitr_1.33        pillar_1.6.2      codetools_0.2-18  reprex_2.0.1     
[65] glue_1.4.2        evaluate_0.14     modelr_0.1.8      vctrs_0.3.8      
[69] tzdb_0.1.2        httpuv_1.6.1      cellranger_1.1.0  gtable_0.3.0     
[73] assertthat_0.2.1  xfun_0.25         broom_0.7.9       later_1.2.0      
[77] viridisLite_0.4.0 globals_0.14.0    ellipsis_0.3.2   </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
